watersheds/_base.py

```diff
@@ -0,0 +1,101 @@
+import geopandas as gpd
+
+
+def construct_basin_dict():
+    full_basin_dict = {}
+    full_parent_basin_dict = {}
+
+    basin_lv12_path = 'watersheds/data/na_basin_lv12.geojson'
```

Comment:
This path won't be available when you install as a pypi artifact.

Data should go in `~/.cache/` - you can steal the downloading utils from:
https://github.com/castorini/pyserini/blob/master/pyserini/util.py

Reply:
Thank you for point out.
I don't have exp on pypi artifact. Need more knowledge here.
So for instances, if we want to include 'na_basin_lv12.geojson' in pypi artifact, what are we going to do? 1 or 2?
1. download it from orginal url, if download, is there way to download it once and save it in ~/.cache/?
2. wrap it into pypi artifact

Reply:
Yes, (1).

Reply:
Since all HydroBASINS and HydroRIVERS need to be converted from shp to geojson. Instead of downloading directly from official dataset, I am thinking upload to our own dropbox. There will be two advantages:
1. We can avoid dataset url be invaild in future
2. We can download converted geojson directly, intstead of downloading shp and convert